File: models/anime.go
package models

import (
	"github.com/Krystian19/cactus-core/proto"
)

// Anime : Anime model
type Anime struct {
	ID        int64
	Title     string
	CreatedAt string
	UpdatedAt string
}

// ToProto : Translates Anime struct into a proto Anime struct
func (a *Anime) ToProto() *proto.Anime {
	return &proto.Anime{
		Id:        a.ID,
		Title:     a.Title,
		CreatedAt: a.CreatedAt,
		UpdatedAt: a.UpdatedAt,
	}
}

// TableName : Sets the model's tablename
func (Anime) TableName() string {
	return "Animes"
}

File: src/db/models/anime.rs
use super::services::proto;
use std::string::String;

pub struct Anime {
  pub id: i64,
  pub title: String,
  pub created_at: std::string::String,
  pub updated_at: std::string::String,
}

pub trait AnimeMethods {
  fn to_proto(&self) -> proto::Anime;
}

impl AnimeMethods for Anime {
  fn to_proto(&self) -> proto::Anime {
    proto::Anime {
      id: self.id,
      title: String::from(&self.title),
      created_at: String::from(&self.created_at),
      updated_at: String::from(&self.updated_at),
    }
  }
}